monte_carlo.py

```diff
@@ -187,7 +198,7 @@ def transport_neutron(materials, bounds, tallies, fission_banks, first_round,
                 
                 # sample for fission event
                 mat = mesh.get_material(neutron.get_cell())
-                if sample_fission(mat) == 1:
+                if sample_fission(mat, neutron.group, new_group) == 1:
```

Comment:
Shouldn't the fission also depend on the material? Since the cross-sections are dependent upon the material?